lib/licensed/sources/npm.rb

```diff
@@ -135,6 +135,14 @@ def yarn_lock_present
       def include_non_production?
         config.dig("npm", "production_only") == false
       end
+
+      def missing_peer?(parent, dependency, name)
+        dependency["peerMissing"] || dependency["missing"] && peer_dependency?(parent, name)
+      end
+
+      def peer_dependency?(parent, name)
+        parent["peerDependencies"] && parent["peerDependencies"][name]
```

Comment:
nit: a bit shorter syntax here with [dig](https://ruby-doc.org/core-3.0.2/Hash.html#method-i-dig)
```suggestion
        parent.dig("peerDependencies", name)
```

Reply:
@jonabc Looks like parent can be a String sometimes... Weird.

![image](https://user-images.githubusercontent.com/633524/152320824-4ffe7f80-8b51-41ae-85ff-48b18ce905aa.png)